kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java

```diff
@@ -0,0 +1,278 @@
+import org.eclipse.kura.rest.configuration.api.DTOUtil;
+import org.eclipse.kura.rest.configuration.api.FailureHandler;
+import org.eclipse.kura.rest.configuration.api.PidSet;
+import org.eclipse.kura.rest.configuration.api.UpdateComponentConfigurationRequest;
+import org.osgi.service.useradmin.Role;
+import org.osgi.service.useradmin.UserAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("networkConfiguration/v1")
+public class NetworkConfigurationRestService {
+
+    private static final Logger logger = LoggerFactory.getLogger(NetworkConfigurationRestService.class);
+
+    private static final String APP_ID = "NETCONF-V1";
+
+    private static final String KURA_PERMISSION_REST_CONFIGURATION_ROLE = "kura.permission.rest.network.configuration";
+
+    private static final String NETWORK_CONF_SERVICE_PID = "org.eclipse.kura.net.admin.NetworkConfigurationService";
+    private static final String IP4_FIREWALL_CONF_SERVICE_PID = "org.eclipse.kura.net.admin.FirewallConfigurationService";
+    private static final String IP6_FIREWALL_CONF_SERVICE_PID = "org.eclipse.kura.net.admin.ipv6.FirewallConfigurationServiceIPv6";
+
+    private static final List<String> NETWORK_CONFIGURATION_PIDS = Arrays.asList(NETWORK_CONF_SERVICE_PID,
+            IP4_FIREWALL_CONF_SERVICE_PID, IP6_FIREWALL_CONF_SERVICE_PID);
+
+    private static final String SNAPSHOT_SUBTASK_ID = "snapshot";
+
+    private final RequestHandler requestHandler = new JaxRsRequestHandlerProxy(this);
+
+    private ConfigurationService configurationService;
+    private CryptoService cryptoService;
+
+    public void setUserAdmin(UserAdmin userAdmin) {
+        userAdmin.createRole(KURA_PERMISSION_REST_CONFIGURATION_ROLE, Role.GROUP);
+    }
+
+    public void setConfigurationService(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    public void setCryptoService(CryptoService cryptoService) {
+        this.cryptoService = cryptoService;
+    }
+
+    public void setRequestHandlerRegistry(final RequestHandlerRegistry registry) {
+        try {
+            registry.registerRequestHandler(APP_ID, this.requestHandler);
+        } catch (final Exception e) {
+            logger.warn("failed to register request handler", e);
+        }
+    }
+
+    public void unsetRequestHandlerRegistry(final RequestHandlerRegistry registry) {
+        try {
+            registry.unregister(APP_ID);
+        } catch (KuraException e) {
+            logger.warn("failed to unregister request handler", e);
+        }
+    }
+
+    /**
+     * GET method.
+     *
+     * Lists the tracked network configurable component Pids
+     *
+     * @return a List of String objects representing the Pids of network factory components
+     *         tracked by the
+     *         {@link ConfigurationService}
+     */
+    @GET
+    @RolesAllowed("network.configuration")
+    @Path("/networkConfigurableComponents")
```

Comment:
Maybe you can replace `networkConfigurableComponents` with `configurableComponents` to be consistend with CONF-V2, we already have `networkConfiguration` in the prefix